### DISCUSSION TITLE
Custom resolutions auto switch possible?

### DISCUSSION THREAD

**Discussion by lastmetroid** [LizardByte/community#271] [Sunshine]
Hello,

im new to Sunshine and was using Nvidia Experience before. Simple question:

How do you manage custom resolutions with Sunshine? For example 2408x1080 for phone resolution. Is there a way, that Sunshine automatically sets the proper resolution, the client sends to use? I cant find any information for this.

For Nvidia streaming, this was not possible, and I had used a workaround for this, that I set 2 batch files on my desktop:

"C:\Programme (frei)\nircmd\nircmd.exe" setdisplay 2408 1080 32
"C:\Programme (frei)\nircmd\nircmd.exe" setdisplay 1920 1080 32

But this just works, if there is a monitor connected to the streaming PC and the monitor is also on and the monitor supports scaling, otherwise you cant set these resolutions on the fly.

Whats the proper way to use this with Sunshine?

Thank you very much

**Comment by ReenigneArcher** [comment]
Better support will be added as part of https://github.com/LizardByte/Sunshine/pull/1430

For now there are alternatives -> https://docs.lizardbyte.dev/projects/sunshine/en/latest/about/app_examples.html#changing-resolution-and-refresh-rate-windows

**Reply by lastmetroid** [reply]
Is there a way to make it automatically run the command if you connect from a phone? for example add 2 different desktop profiles:

Desktop 1080
Desktop Custom 2408

For example and then the profile executes the qres.exe on connect? With that "Elevating Commands (Windows)" way? isnt it critical the resolution needs to be switched before the client connects or is that irrelevant.

**Reply by ReenigneArcher** [reply]
Prep commands should run before the stream is established.

**Reply by lastmetroid** [reply]
Thanks I just tested it and it seems to work. but there is a limitation, that Nvidia seems to ignore the resolution command if the monitor which is connected doesnt support it, is there a work around for that?

**Reply by ReenigneArcher** [reply]
Probably a dummy display dongle, with all the resolutions you want added to it's bin.

**Reply by lastmetroid** [reply]
Is there one you can program with custom resolutions? I have actually one from Amazon, but you cant program it. and it is missing all the phone resolutions, just some default ones.

**Reply by ReenigneArcher** [reply]
@Nonary can you offer some suggestions?

**Reply by Nonary** [reply]
There's only one besides the NVIDIA/Amd drivers and that is CRU, it's not exactly user friendly so I would suggest checkout out YouTube videos.

**Reply by ReenigneArcher** [reply]
https://discord.com/channels/352065098472488960/465369864534491146/1117725570533429270

**Reply by lastmetroid** [reply]
Thanks. Ive looked into CRU actually and indeed, it is very very user unfriendly. Wish there was some auto mode, where you could just enter x y and maybe refresh rate and the rest would be entered automatically. Have to look into it how I can use that CRU tool. With Nvidia control panel it seems to be easier, but with AMD tool, it wont work, it will always claim the monitor is not supporting the custom resolution.

[a comment by an automated account is not shown]
